vertx-web/src/main/java/io/vertx/ext/web/handler/impl/BodyHandlerImpl.java

```diff
@@ -160,30 +179,35 @@ public BHandler(RoutingContext context, long contentLength) {
       initBodyBuffer(contentLength);
 
       if (isMultipart || isUrlEncoded) {
-        makeUploadDir(context.vertx().fileSystem());
         context.request().setExpectMultipart(true);
-        context.request().uploadHandler(upload -> {
-          if (bodyLimit != -1 && upload.isSizeAvailable()) {
-            // we can try to abort even before the upload starts
-            long size = uploadSize + upload.size();
-            if (size > bodyLimit) {
-              failed = true;
-              context.fail(413);
-              return;
+        if (handleFileUploads) {
+          makeUploadDir(context.vertx().fileSystem());
+          context.request().uploadHandler(upload -> {
+            if (bodyLimit != -1 && upload.isSizeAvailable()) {
+              // we can try to abort even before the upload starts
+              long size = uploadSize + upload.size();
+              if (size > bodyLimit) {
+                failed = true;
+                context.fail(413);
+                return;
+              }
             }
-          }
-          // we actually upload to a file with a generated filename
-          uploadCount.incrementAndGet();
-          String uploadedFileName = new File(uploadsDir, UUID.randomUUID().toString()).getPath();
-          upload.streamToFileSystem(uploadedFileName);
-          FileUploadImpl fileUpload = new FileUploadImpl(uploadedFileName, upload);
-          fileUploads.add(fileUpload);
-          upload.exceptionHandler(t -> {
-            deleteFileUploads();
-            context.fail(t);
+            // we actually upload to a file with a generated filename
+            uploadCount.incrementAndGet();
+            String uploadedFileName = new File(uploadsDir, UUID.randomUUID().toString()).getPath();
+            upload.streamToFileSystem(uploadedFileName);
+            FileUploadImpl fileUpload = new FileUploadImpl(uploadedFileName, upload);
+            fileUploads.add(fileUpload);
+            upload.exceptionHandler(t -> {
+              deleteFileUploads();
+              context.fail(t);
+            });
+            upload.endHandler(v -> uploadEnded());
           });
-          upload.endHandler(v -> uploadEnded());
-        });
+        } else {
+          // Add a NOOP upload handler, to ignore files.
+          context.request().uploadHandler(NOOP_UPLOAD_HANDLER);
```

Comment:
I'm not sure about this. We might still want to limit/block when uploads are larger than a safe limit. This could be the door for a DDoS. So pretty much it should be like before but without writing to disk

Reply:
The complete check is done in `handle(Buffer)` at L250; I added a new test to prove it (testBodyLimitWithHandleFileUploadsFalse). To me it looked like the ckeck in here was to avoid writing to disk if: 
* a single part of the multipart data is bigger than the max body size
* the part declares it's size via a "Content-Length" header (checked via `upload.isSizeAvailable()`)

I thought that, sice we are not accumulating the bytes in memory but just discarding them, we could ignore this quick fail because, proably, an attacker would not declare the content length anyway.

If you prefer to have the quick fail anyway, just let me know and I can surelly do that.

Reply:
I think the quick fail check doesn't hurt, right?